protocol/x/clob/keeper/process_operations.go

```diff
@@ -675,6 +675,34 @@ func (k Keeper) PersistMatchDeleveragingToState(
 	}
 	deltaBaseQuantumsIsNegative := position.GetIsLong()
 
+	// If there are zero-fill deleveraging operations, this is a sentinel value to indicate a subaccount could not be
+	// liquidated or deleveraged and still has negative equity. Mark the current block number in state to indicate a
+	// negative TNC subaccount was seen.
+	if len(matchDeleveraging.GetFills()) == 0 {
+		if !shouldDeleverageAtBankruptcyPrice {
+			return errorsmod.Wrapf(
+				types.ErrZeroFillDeleveragingForNonNegativeTncSubaccount,
+				fmt.Sprintf(
+					"PersistMatchDeleveragingToState: zero-fill deleveraging operation included for subaccount %+v"+
+						" and perpetual %d but subaccount isn't negative TNC",
+					liquidatedSubaccountId,
+					perpetualId,
+				),
+			)
+		}
+
+		telemetry.IncrCounterWithLabels(
```

Comment:
can you use the new metrics entrypoint [here](https://github.com/dydxprotocol/v4-chain/blob/main/protocol/lib/metrics/util.go) instead?